# Exercise 4. Variables and Names

If you get stuck with this exercise, remember the tricks you have been taught so far of finding differences and focusing on details:
1. Write a comment above each line explaining to yourself in English what it does.
2. Read your .py file backward.
3. Read your .py file out loud, saying even the characters.

In [3]:
cars = 100
space_in_a_car=4.0
drivers=30
passengers=90
cars_not_driven=cars-drivers
cars_driven=drivers
carpool_capacity =cars_driven * space_in_a_car
average_passengers_per_car =passengers/cars_driven
print("There are",cars,"cars available")
print("There are only",drivers,"drivers available")
print("There will be ",cars_not_driven,"empty cars today.")
print("We can transpot",carpool_capacity,"people today.")
print("We have",passengers,"to carpool today")
print("We need to put about",average_passengers_per_car,"in each car.")

There are 100 cars available
There are only 30 drivers available
There will be  70 empty cars today.
We can transpot 120.0 people today.
We have 90 to carpool today
We need to put about 3.0 in each car.


Here are more study drills:

1. I used 4.0 for space_in_a_car, but is that necessary? What happens if it’s just 4?

We can transpot 120.0 people today. ->We can transpot 120 people today.

2. Remember that 4.0 is a floating point number. It’s just a number with a decimal point, and you need 4.0 instead of just 4 so that it is floating point.

3. Write comments above each of the variable assignments.
NO

4. Make sure you know what = is called (equals) and that its purpose is to give data (numbers, strings, etc.) names (cars_driven, passengers).

5. Remember that _ is an underscore character.

6. Try running python3.6 from the Terminal as a calculator like you did before, and use variable names to do your calculations. Popular variable names include i, x, and j.

## Note
Can we write x=100 instead of x = 100? You can, but it’s bad form. You should add space around operators like this so that it’s easier to read.

What do you mean by “read the file backward”? Very simple. Imagine you have a file with 16 lines of code in it. Start at line 16, and compare it to my file at line 16. Then do it again for 15, and so on until you’ve read the whole file backward.